salt/modules/mac_shadow.py

```diff
@@ -30,95 +33,99 @@ def __virtual__():
     return __virtualname__
 
 
-def _run_parse_return_value(cmd):
+def _get_account_policy(name):
     '''
-    Execute the command, parse the return, return the value
-    For use by this module only
+    Get the entire accountPolicy and return it as a dictionary. For use by this
+    module only
 
-    Args:
-        cmd: a dscl command to run
+    :param str name: The user name
 
-    Returns:
-        the value found in the command
+    :return: a dictionary containing all values for the accountPolicy
+    :rtype: bool
 
+    :raises: CommandExecutionError on user not found or any other unknown error
     '''
-    ret = __salt__['cmd.run'](cmd)
-    if ': ' in ret:
-        value = ret.split(': ')[1]
-        return value
-    if ':\n' in ret:
-        value = ret.split(':\n')[1]
-        return value
-    return ret
 
+    cmd = 'pwpolicy -u {0} -getpolicy'.format(name)
+    try:
+        ret = salt.utils.mac_utils.execute_return_result(cmd)
+    except CommandExecutionError as exc:
+        if 'Error: user <{0}> not found'.format(name) in exc.strerror:
+            raise CommandExecutionError('User not found: {0}'.format(name))
+        raise CommandExecutionError('Unknown error: {0}'.format(exc.strerror))
 
-def _get_dscl_data_value(name, key):
+    try:
+        policy_list = ret.split('\n')[1].split(' ')
+        policy_dict = {}
+        for policy in policy_list:
+            if '=' in policy:
+                key, value = policy.split('=')
+                policy_dict[key] = value
+        return policy_dict
+    except IndexError:
+        return {}
+
+
+def _set_account_policy(name, policy):
     '''
-    Return the value for a key in the user's plist file
-    For use by this module only
+    Set a value in the user accountPolicy. For use by this module only
 
-    Args:
-        name: username
-        key: plist key
+    :param str name: The user name
+    :param str policy: The policy to apply
 
-    Returns:
-        the value contained within the key
+    :return: True if success, otherwise False
+    :rtype: bool
+
+    :raises: CommandExecutionError on user not found or any other unknown error
     '''
-    cmd = 'dscl . -read /Users/{0} {1}'.format(name, key)
-    return _run_parse_return_value(cmd)
+    cmd = 'pwpolicy -u {0} -setpolicy "{1}"'.format(name, policy)
+
+    try:
+        return salt.utils.mac_utils.execute_return_success(cmd)
+    except CommandExecutionError as exc:
+        if 'Error: user <{0}> not found'.format(name) in exc.strerror:
+            raise CommandExecutionError('User not found: {0}'.format(name))
+        raise CommandExecutionError('Unknown error: {0}'.format(exc.strerror))
```

Comment:
It will either return a CommandExecutionError with 'User not found: <user_name>' as the message or 'Unknown error: <error_message>' on any other error.